### PR TITLE
Use nullptr, not 0 in the C++ API impl.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
This time, it also includes `stddef.h`, so it should work everywhere.

### CODE REVIEW

**Comment by waywardmonkeys** [comment]
This is failing because we need to build the examples with C++11. I'll keep at it.

File: examples/tptp/CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,22 @@ find_package(Z3
   # use this option.
   NO_DEFAULT_PATH
 )
+
+################################################################################
```
Comment (NikolajBjorner) [inline examples/tptp/CMakeLists.txt]: they didn't before. I don't know if requiring C++11 here is an advantage: what if clients forget to require C++11. They their builds fail.

Comment (waywardmonkeys) [reply examples/tptp/CMakeLists.txt]: We've been building the other examples with C++11 (I copied that from the `examples` `CMakeLists.txt`) ... also, in the 4.6 release notes, it indicates that C++11 is required for both building Z3 and using the C++ API.

Comment (waywardmonkeys) [reply examples/tptp/CMakeLists.txt]: We could put a check for the C++ version in `z3++.h` and issue a warning / error if it is less than C++11?

**Comment by NikolajBjorner** [comment]
This would be adding extra requirements to clients and I don't see what would justify these requirements.

**Comment by waywardmonkeys** [comment]
I was going to follow this up with other usages of C++11 to address some issues with copying of data ... but I don't actually use the C++ API (or Java or dotnet or Python), so I'll probably hold off then. (I sent you a direct email a while ago about my rough directions.)

**Pull request closed without merging**